bundles/org.openhab.binding.ism8/src/main/java/org/openhab/binding/ism8/internal/Ism8Handler.java

```diff
@@ -0,0 +1,167 @@
+import org.openhab.binding.ism8.server.IDataPoint;
+import org.openhab.binding.ism8.server.IDataPointChangeListener;
+import org.openhab.binding.ism8.server.Server;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link Ism8Handler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Hans-Reiner Hoffmann - Initial contribution
+ */
+@NonNullByDefault
+public class Ism8Handler extends BaseThingHandler implements IDataPointChangeListener {
+    private final Logger logger = LoggerFactory.getLogger(Ism8Handler.class);
+
+    private @Nullable Ism8Configuration config;
+    private @Nullable Server server;
+
+    public Ism8Handler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        this.logger.debug("Ism8: Handle command = {} {}", channelUID.getId(), command);
+        Channel channel = getThing().getChannel(channelUID);
+        Server svr = this.server;
+        if (channel != null && svr != null) {
+            if (channel.getConfiguration().containsKey("id") && channel.getConfiguration().containsKey("write")
+                    && channel.getConfiguration().get("write").toString().equalsIgnoreCase("true")) {
+                IDataPoint dataPoint = null;
+                try {
+                    int id = Integer.parseInt(channel.getConfiguration().get("id").toString());
+                    this.logger.debug("Channel '{}' writting into ID '{}'", channel.getUID().getId(), id);
+                    this.updateState(channelUID, new QuantityType<>(command.toString()));
+
+                    for (IDataPoint dp : svr.getDataPoints()) {
+                        if (dp.getId() == id) {
+                            dataPoint = dp;
+                            break;
+                        }
+                    }
+                } catch (NumberFormatException e) {
+                    this.logger.debug("Updating State of ISM DataPoint '{}' failed. '{}'", channel.getConfiguration(),
+                            e.getMessage());
+                }
+
+                if (dataPoint != null) {
+                    try {
+                        svr.sendData(dataPoint.createWriteData(command));
+                    } catch (IOException e) {
+                        this.logger.debug("Writting to ISM DataPoint '{}' failed. '{}'", dataPoint.getId(),
+                                e.getMessage());
+                    }
+                }
+            }
+        }
+    }
+
+    @Override
+    public void dispose() {
+        Server svr = this.server;
+        if (svr != null) {
+            svr.stopServerThread();
+        }
+    }
+
+    @Override
+    public void initialize() {
+        this.config = getConfigAs(Ism8Configuration.class);
+        Ism8Configuration cfg = this.config;
+        final String uid = this.getThing().getUID().getAsString();
+        if (cfg != null) {
```

Comment:
`cfg` cannot be null at this point. You could remove the null check.

Reply:
OK, changed.